### PR TITLE
Request mock should handle only specified requests (part of #2336)

### PR DESCRIPTION
**Pull request by miherlosev**
@AndreyBelym

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
FPR @AndreyBelym

File: src/api/request-hooks/request-mock.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { RequestHookConfigureAPIError } from '../../errors/test-run/index';
 
 class RequestMock extends RequestHook {
     constructor () {
-        super();
+        super(null);
```
Comment (AndreyBelym) [inline src/api/request-hooks/request-mock.js]: Why we can't just pass an empty array here?

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Pull request merged**